src/mongoc/mongoc-topology.c

```diff
@@ -351,29 +359,23 @@ mongoc_topology_select (mongoc_topology_t         *topology,
          topology->stale = false;
       }
 
-      /* until we find a server or timeout */
-      for (;;) {
```

Comment:
Right -- that retry-logic is the root of SPEC-217.
The coolDown SDAM option was for subsequent user triggered actions, which tweaks _mongoc_topology_time_to_scan() decision to call for a scan or not.